sw/host/opentitanlib/src/transport/hyperdebug/uart.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+use std::time::Duration;
+use std::io::Read;
+use std::io::Write;
+
+use crate::io::uart::Uart;
+use crate::transport::hyperdebug::Error;
+use crate::transport::hyperdebug::Hyperdebug;
+
+pub struct HyperdebugUart {
+    port: Box<dyn serialport::SerialPort>,
+}
+
+impl HyperdebugUart {
+    pub fn open(hyperdebug: &Hyperdebug) -> Result<Self> {
+        let port =
+            serialport::new(hyperdebug.uart_ttys[0].to_str().ok_or(Error::UnicodePathError)?,
+                            115_200)
+            .timeout(Duration::from_millis(100))
+            .open().expect("Failed to open port");
+        let result = HyperdebugUart {
+            port,
+        };
+        Ok(result)
+    }
+}
+
+impl Write for HyperdebugUart {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.port.write(buf)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+impl Read for HyperdebugUart {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        self.port.set_timeout(Duration::MAX)?;
+        self.port.read(buf)
+    }
+}
+
+impl Uart for HyperdebugUart {
+    fn get_baudrate(&self) -> u32 {
+        match self.port.baud_rate() {
+            Ok(baud_rate) => baud_rate,
+            _ => panic!("SerialPort::baud_rate() returned Err")
```

Comment:
maybe `expect()` or `unwrap()` is what you want here? Seems a bit odd to panic, though.